apps/condo/domains/common/utils/serverSchema/perFieldAccess.js

```diff
@@ -0,0 +1,94 @@
+const { transformCRUDString } = require('@condo/domains/common/utils/serverSchema/accessUtils')
+const { GQL_LIST_SCHEMA_TYPE, GQL_CUSTOM_SCHEMA_TYPE, getSchemaCtx } = require('@core/keystone/schema')
+const { get } = require('lodash')
+
+function getFieldPermissions (user, listKey, field) {
+    return transformCRUDString(get(user, ['permissions', 'lists', listKey, 'fields', field], ''))
+}
+
+function getCustomPermissions (user, key) {
+    return get(user, ['permissions', 'custom', key], null)
+}
+
+function hasDefinedPermissions (user) {
+    return get(user, 'permissions') != null
+}
+
+function denyIfHasNoPermission (user, permission, originalAccessResult) {
+    if (!originalAccessResult) return false
+    if (hasDefinedPermissions(user)) {
+        return permission === true
+    }
+    return true
+}
+
+function perFieldAccess (schemaType, schemaName, schema) {
+    if (schemaType === GQL_LIST_SCHEMA_TYPE) {
+        Object.keys(schema.fields).forEach(field => {
```

Comment:
It looks too complex if you add an access function to all fields. Do we really need it?

Reply:
Yes if you want permissions-based rights model